alert/tests.py

```diff
@@ -532,21 +534,341 @@ def test_no_permission(self):
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_update_settings_change_first_name(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob', 'first_name': 'newname', 'last_name': '',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], 'newname')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], 'newname')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_update_settings_change_last_name(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob', 'first_name': '', 'last_name': 'newname',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], 'newname')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], 'newname')
+
+    def test_update_settings_change_username(self):
```

Comment:
Usernames are set by Platform to be pennkeys, so users shouldn't be able to change it with an API request. I'd be worried someone changes their username to an unregistered user so that when that user logs in the first time, they're not able to access the site (or worse, they're logged in to an existing account)